src/qentry.c

```diff
@@ -740,7 +740,7 @@ static bool _save(qentry_t *entry, const char *filepath)
     if (entry == NULL) return false;
 
     FILE *fd;
-    if ((fd = fopen(filepath, "w")) < 0) {
+    if ((fd = fopen(filepath, "w")) == NULL) {
```

Comment:
Agree. I'm merging this code now, further improvement can be handled in a separate pr.
Thanks